tcms/testruns/tests/test_permissions.py

```diff
@@ -0,0 +1,80 @@
+from django.urls import reverse
+from http import HTTPStatus
+
+from tcms import tests
+from tcms.tests import factories
+from tcms.testcases.models import TestCaseStatus
+from tcms.testruns.models import TestRun, TestExecutionStatus
+
+
+class CreateTestRunTestCase(tests.PermissionsTestCase):
+    permission_label = 'testruns.add_testrun'
+    url = reverse('testruns-new')
+    http_method_names = ['get', 'post']
+
+    @classmethod
+    def setUpTestData(cls):
+        plan = factories.TestPlanFactory()
+        build = factories.BuildFactory()
+
+        cls.post_data = {
+            'summary': plan.name,
+            'from_plan': plan.name,
+            'build': build.pk,
+            'notes': 'Create new test run',
+            'POSTING_TO_CREATE': 'YES',
+        }
+
+        super().setUpTestData()
+
+        # cls.tester is created after calling super() above
+        cls.post_data['manager'] = cls.tester.email
+        cls.post_data['default_tester'] = cls.tester.email
+
+        case_status_confirmed = TestCaseStatus.get_confirmed()
+
+        case_1 = factories.TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=case_status_confirmed,
+            plan=[plan])
+        case_1.save()  # will generate history object
+
+        case_2 = factories.TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=case_status_confirmed,
+            plan=[plan])
+        case_2.save()  # will generate history object
+
+        cls.post_data['case'] = (case_1.pk, case_2.pk)
+
+    def verify_post_with_permission(self):
+        new_run = TestRun.objects.last()
```

Comment:
Two things here:

1) When moving code around delete the origin in `testruns/tests/test_views.py::TestCreateNewRun.test_create_a_new_run()`, otherwise you end up with duplciates

2) The new test case is not a verbatim copy of the old. A simple copy &  paste here doesn't work, you have to think about what are the important things we want to be checking(aka verifying) in this scenario. Again analyze `bugs/tests/test_permissions.py::NewTestCase.verify_post_with_permission` for hints.